app/services/extension.js

```diff
@@ -135,9 +137,19 @@ export default Service.extend({
 
   /* Schema Editor */
   validateSchema() {
+    let allValid = true;
+
     get(this, 'data._schema').forEach(schemaField => {
-      set(schemaField, 'validation', validateSchemaField(schemaField));
+      const validation = validateSchemaField(schemaField);
+      set(schemaField, 'validation', validation);
+      if (validation.length > 0) allValid = false;
     });
```

Comment:
fyi, this is what javascript's "every" is for:

```
const allValid = get(this, 'data._schema').every(schemaField => (get(schemaField, 'validation') || []).length === 0))
```

whenever you find yourself assigning an "outside" variable inside of a `forEach`, there's almost definitely a cleaner way to do it!

that said - my suggestion is technically slower than what you've done (two loops instead of one), but i think it's worth it for readability.

you can also do:

```
    const allValid = get(this, 'data._schema').map(schemaField => {
      set(schemaField, 'validation', validateSchemaField(schemaField));
      const validation = validateSchemaField(schemaField);
      set(schemaField, 'validation', validation);
      return (validation.length === 0)
    }).every(Boolean);
```

or you can use `some` (but every is more readable in this case)

```
    const someInvalid = get(this, 'data._schema').map(schemaField => {
      set(schemaField, 'validation', validateSchemaField(schemaField));
      const validation = validateSchemaField(schemaField);
      set(schemaField, 'validation', validation);
      return (validation.length > 0)
    }).some(Boolean);
```